tests/test_macos.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+import os
+from typing import Any
+
+import pytest
+
+from platformdirs.macos import MacOS
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        {},
+        {"appname": "foo"},
+        {"appname": "foo", "version": "v1.0"},
+    ],
+    ids=[
+        "no_args",
+        "app_name",
+        "app_name_version",
+    ],
+)
+def test_macos(params: dict[str, Any], func: str) -> None:
+    result = getattr(MacOS(**params), func)
+
+    home = os.path.expanduser("~")
+    suffix_elements = []
+    if "appname" in params:
+        suffix_elements.append(params["appname"])
+    if "version" in params:
+        suffix_elements.append(params["version"])
+    if suffix_elements:
+        suffix_elements.insert(0, "")
+    suffix = "/".join(suffix_elements)
+
+    expected_map = {
+        "user_data_dir": f"{home}/Library/Application Support{suffix}",
+        "site_data_dir": f"/Library/Application Support{suffix}",
+        # Please note that the config dirs are NOT */Library/Preferences!
+        # For details see: https://github.com/platformdirs/platformdirs/issues/98
```

Comment:
guess this should avoid somebody changing this back to the wrong dirs again.

Reply:
Can you remove this comment? The test itself is enough with the git blame.

Reply:
Considering this was initially correct, then broken for platformdirs 2.0.0 and now back in the initial correct state, I would strongly suggest we keep that comment, so that this never happens again.

Reply:
I disagree and I'ld appreciate another opinion here.

You can't just assume that the next guy who breaks this will do git blame or a lot of other research first. Otherwise the breakage at 2.0.0 would also not have happened.

And if such a comment would have been there before the 2.0.0 breakage, we would not have had all 2.x.x release wrong on macOS and also would not have had 2 incompatible/breaking releases in total for macOS users.